vantage/business_metric_resource_model.go

```diff
@@ -0,0 +1,258 @@
+package vantage
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	modelsv2 "github.com/vantage-sh/vantage-go/vantagev2/models"
+)
+
+type businessMetricResourceModel struct {
+	CostReportTokensWithMetadata types.List   `tfsdk:"cost_report_tokens_with_metadata"`
+	CreatedByToken               types.String `tfsdk:"created_by_token"`
+	Title                        types.String `tfsdk:"title"`
+	Token                        types.String `tfsdk:"token"`
+	Values                       types.List   `tfsdk:"values"`
+}
+
+type businessMetricResourceModelValue struct {
+	Amount types.Float64 `tfsdk:"amount"`
+	Date   types.String  `tfsdk:"date"`
+}
+
+type businessMetricDataSourceModelValue struct {
+	Amount types.String `tfsdk:"amount"`
+	Date   types.String `tfsdk:"date"`
+}
+type businessMetricResourceModelCostReportToken struct {
+	CostReportToken types.String `tfsdk:"cost_report_token"`
+	UnitScale       types.String `tfsdk:"unit_scale"`
+}
+
+func (m *businessMetricResourceModel) applyPayload(ctx context.Context, payload *modelsv2.BusinessMetric, isDataSource bool) diag.Diagnostics {
```

Comment:
isDataSource is passed to `parseValues` so that it can know when the expected input is a float vs a string. Floats are used in value amounts of the Create* model, whereas strings are used in the data source model.